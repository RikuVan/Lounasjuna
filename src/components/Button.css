.Button {
  background-color: rgba(255, 255, 255, 0.1);
  border: 1px solid #aa8899;
  border-radius: 3px;
  color: #aa8899;
  font-size: 1rem;
  outline: none;
  padding: 10px 15px;
}

.Button-white {
  color: #fff;
}

.Button:hover {
  opacity: 0.9;
}

.Button:active {
  transform: scale(0.98);
}
