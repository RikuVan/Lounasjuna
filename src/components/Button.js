import React, {PropTypes} from 'react'

const Button = ({type, onClick, text}) => (
  <button
    className={type ? type : 'default'}
    onClick={onClick}>
    {text}
  </button>
)

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired
}

export default Button