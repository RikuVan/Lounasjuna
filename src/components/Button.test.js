import React from 'react'
import Button from './Button'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

test('Button snapshot test', () => {
  console.error = jest.genMockFn();
  const component = shallow(<Button />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})