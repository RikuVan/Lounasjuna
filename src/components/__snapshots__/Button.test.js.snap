exports[`test Button snapshot test 1`] = `
<button
  className="Button  undefined" />
`;
